src/monster.cpp

```diff
@@ -1729,6 +1731,10 @@ void monster::process_effects()
                     made_of("wool"))
                     apply_damage( nullptr, bp_torso, rng( 15, 40 ) );
             }
+            if( id == "shrieking"){
+                 sounds::sound(pos(), 120, _("a piercing wail!"), true);
+                 moves -= 40;
+            }
```

Comment:
Implementing it as an effect checked for every monster sounds hacky.
I think it would be better to add another attack (say, `shriek_continue`) which would have no cooldown and would proc this wail. It would also prevent the cases where a shrieker polymorphed into a blob/fungus/etc. keeps shrieking.